js/wbf.js

```diff
@@ -0,0 +1,616 @@
+        //     "bid_id":1001,
+        //     "ask_id":1002,
+        //     "bid_user_id":10001,
+        //     "ask_user_id":10001
+        // }
+        const timestamp1 = this.safeString (order, 'ctime');
+        const timestamp2 = this.safeString (order, 'created_at');
+        const timestamp = timestamp1 || timestamp2;
+        return {
+            'id': this.safeString (order, 'id'),
+            'datetime': this.iso8601 (timestamp),
+            'timestamp': timestamp,
+            'lastTradeTimestamp': undefined,
+            'status': this.safeString (order, 'status_msg'),
+            'symbol': market['symbol'],
+            'side': this.safeValue (order, 'side'),
+            'type': this.safeString (order, 'type'),
+            'price': this.safeFloat (order, 'price'),
+            'cost': undefined,
+            'amount': this.safeFloat (order, 'volume'),
+            'filled': this.safeValue (order, 'deal_volume'),
+            'remaining': this.safeValue (order, 'remain_volume'),
+            'fee': this.safeValue (order, 'fee'),
+            'info': order,
+        };
+    }
+
+    nonce () {
+        return this.milliseconds ();
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'][api] + '/' + this.implodeParams (path, params);
+        const url2 = '/open/api/' + this.implodeParams (path, params);
+        let query = this.omit (params, this.extractParams (path));
+        if (method === 'GET') {
+            if (Object.keys (query).length) {
+                url += '?' + this.urlencode (query);
+            }
+        }
+        if (api === 'private') {
+            this.checkRequiredCredentials ();
+            const timestamp = this.nonce () + '';
+            query['time'] = timestamp;
+            query['api_key'] = this.apiKey;
+            query = this.urlencode (query);
+            const apiKey = this.encode (this.apiKey);
+            const queryArray = query.split ('&');
+            const sortedQuery = this.sortBy (queryArray, 0).join ('&');
+            const auth = method + '\n' + 'openapi.wbf.live' + '\n' + url2 + '\n' + sortedQuery;
+            const signature = this.hmac (this.encode (auth), this.encode (this.secret), 'sha256', 'base64');
+            if (method === 'POST') {
+                url = this.urls['api'][api] + '/' + this.implodeParams (path, params);
+                body = query;
+                body['api_key'] = apiKey;
+                body['time'] = timestamp;
+                body['sign'] = signature;
+            } else {
+                query = this.urlencode ({
+                    'api_key': apiKey,
+                    'time': timestamp,
+                    'sign': signature,
+                });
+                if (Object.keys (params).length) {
+                    url += '&' + query;
+                } else {
+                    url += '?' + query;
+                }
```

Comment:
```suggestion
                symbol = Object.keys(params).length ? '&' : '?';
                url += `{symbol}{query}`;
```

Reply:
@lesterkimkraken that will not transpile to other languages: 

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-contribute-code
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#multilanguage-support
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes

Reply:
Ah, thanks for the background.